ambry-account/src/main/java/com/github/ambry/account/InMemoryAccountService.java

```diff
@@ -0,0 +1,146 @@
+import java.nio.charset.StandardCharsets;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.Collection;
+import java.util.concurrent.atomic.AtomicBoolean;
+import java.util.stream.Collectors;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+
+/**
+ * Account service which holds account data in-memory.
+ *
+ * This implementation of the account service is meant primarily for local testing and deployment.
+ */
+public final class InMemoryAccountService extends AbstractAccountService {
+
+  static Account UNKNOWN_ACCOUNT = new Account(Account.UNKNOWN_ACCOUNT_ID, Account.UNKNOWN_ACCOUNT_NAME, Account.AccountStatus.ACTIVE,
+      Account.ACL_INHERITED_BY_CONTAINER_DEFAULT_VALUE, Account.SNAPSHOT_VERSION_DEFAULT_VALUE,
+      Arrays.asList(Container.UNKNOWN_CONTAINER, Container.DEFAULT_PUBLIC_CONTAINER,
+          Container.DEFAULT_PRIVATE_CONTAINER), Account.QUOTA_RESOURCE_TYPE_DEFAULT_VALUE);
+  private static final Logger logger = LoggerFactory.getLogger(InMemoryAccountService.class);
+
+  /** Location of the account file which this service uses as it's source of accounts.*/
+  private final Path accountFile;
+
+  /**
+   * Constructs an {@code InMemoryAccountService}. {@code #init} method MUST be called after construction.
+   *
+   * @param accountServiceMetrics Metrics instance to log metrics to.
+   */
+  public InMemoryAccountService(Path accountFile, AccountServiceMetrics accountServiceMetrics, AccountServiceConfig accountConfig) {
+    super(accountConfig, accountServiceMetrics, null);
+    this.accountFile = accountFile;
+  }
+
+  /** Guards against calling {@link #init} multiple times.*/
+  private final AtomicBoolean initialized = new AtomicBoolean();
+
+  /**
+   * Initializes this instance. Does all work which requires handing out references to this instance itself which can't
+   * be done during normal instance construction without letting a reference to an incomplete constructed instance escape.
+   */
+  public void init() {
+    if (!initialized.getAndSet(true)) {
+      processAccountJsonFile();
+
+      // add the default unknown account
+      ArrayList<Account> accounts = new ArrayList<Account>();
+      accounts.add(UNKNOWN_ACCOUNT);
+
+      try {
+        updateAccounts(accounts);
+      } catch (AccountServiceException e) {
+        logger.warn("Unable to add default unknown account", e);
+      }
+
+      }
+  }
+
+  @Override
+  protected void checkOpen() {
+    if (!initialized.get()) {
+      throw new IllegalStateException("AccountService is closed.");
+    }
+  }
+
+  @Override
+  protected void onAccountChangeMessage(String topic, String message) {
+    // We're good.
+  }
+
+  /**
+   * Adds accounts
+   *
+   * @param accounts The collection of {@link Account}s to update. Cannot be {@code null}.
+   */
+  @Override
+  public void updateAccounts(Collection<Account> accounts) throws AccountServiceException {
```

Comment:
Usually we would check if there are duplicates in the provided accounts list before checking the duplicates against the existing accounts. It's weird that this logic is not in the PostAccountsHandler, but in each account service implementation. However, let's follow that convention by calling `AccountUtils.hasDuplicateAccountIdOrName` here.